src/components/atoms/VideoOverlayButton/VideoOverlayButton.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect } from "react";
+import { useToggle } from "react-use";
+import {
+  faMicrophone,
+  faMicrophoneSlash,
+  faVideo,
+  faVideoSlash,
+  faVolumeMute,
+  faVolumeUp,
+  IconDefinition,
+} from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import classNames from "classnames";
+
+import { ContainerClassName } from "types/utility";
+
+import { assertUnreachable } from "utils/error";
+
+import "./VideoOverlayButton.scss";
+
+type VideoOverlayButtonVariant = "microphone" | "camera" | "audio";
+
+export interface VideoOverlayButtonProps extends ContainerClassName {
+  variant: VideoOverlayButtonVariant;
+  defaultValue: boolean;
+  onEnabledChanged: (enabled: boolean) => void;
+}
+
+export const VideoOverlayButton: React.FC<VideoOverlayButtonProps> = ({
+  variant,
+  defaultValue,
+  onEnabledChanged,
+  containerClassName,
+}) => {
+  const [enabled, toggle] = useToggle(defaultValue);
+
+  useEffect(() => {
+    onEnabledChanged(enabled);
+  }, [enabled, onEnabledChanged]);
+
+  let iconEnabled: IconDefinition = faMicrophone;
+  let iconDisabled: IconDefinition = faMicrophoneSlash;
+  switch (variant) {
+    case "microphone":
+      iconEnabled = faMicrophone;
+      iconDisabled = faMicrophoneSlash;
+      break;
+    case "camera":
+      iconEnabled = faVideo;
+      iconDisabled = faVideoSlash;
+      break;
+    case "audio":
+      iconEnabled = faVolumeUp;
+      iconDisabled = faVolumeMute;
+      break;
+    default:
+      assertUnreachable(variant);
+  }
```

Comment:
Switch... not the best structure in the programming languages that can do:

```ts
const {iconEnabled,IconDisabled} = OVERLAY_BUTTON_ICONS[variant];

if(!iconEnabled || !iconDisabled){
    throw Error();
}
```

Reply:
`switch` is perfectly fine and in this case, it doesn't even need runtime error handling as it's compile-time protected.

Reply:
`switch` is never perfectly fine, it's a multiple `goto`s in disguise, and a half way between overengineered simpler approach or underengineered factory

Then why?

```ts
default:
      assertUnreachable(variant);
```

Reply:
This ensures during compilation that switch exhausted all the cases.

Reply:
There is no need if you don't use `switch`. The rule of thumb I have is: 

consider `switch` as a code smell and you will almost always find better way

Example: https://github.com/sparkletown/sparkle/pull/2231#discussion_r701614999

Reply:
I see that you've done a fair amount of work to prove your point
Might as well finish this PR yourself as you see fit.

Reply:
> I see that you've done a fair amount of work to prove your point
@dmitrymarchuk It is not about proving points. It's about explaining my suggestions in clearer way, a rare occasion that involved 15 minutes of coding for the diff to explain better what words couldn't

> Might as well finish this PR yourself as you see fit.
Unfortunately, with time constraints and other issues at hand, I will not have time to build up the context of the issue in general, hence my comments are about code style, not if the PR satisfies the issue requirements in all important cases.

Reply:
Why? With tight time constraints, you managed to find time for a separate PR to provide examples for all your suggestions here.
You clearly will find time to just check out this branch and implement this yourself.

Reply:
There is no need to continue the same conversation from slack DM here. I will summarize it and consider his issues as addressed:

I've suggested  @dmitrymarchuk get another approval and explained it is not an exception to do local coding while reviewing a PR, even if rarely I've pushed code as example (one other case is https://github.com/sparkletown/sparkle/pull/2134#issuecomment-910044232 )

Despite my rationale of this PR might be delayed due to other considerations https://github.com/sparkletown/sparkle/pull/2221#discussion_r701625967
@dmitrymarchuk's decision was to be adamant in me taking over this PR.

That being said, I was OK if this PR got approval from another and I am if another person takes it over to push it over the finish line.

Reply:
Yeah, you are right about the delay, so after discussion with @sunny-viktoryia I will finish this as she will do another review. Thank you for your time on this.